SimTKcommon/Geometry/include/SimTKcommon/internal/DecorativeGeometry.h

```diff
@@ -596,6 +596,79 @@ class SimTK_SimTKCOMMON_EXPORT DecorativeMeshFile : public DecorativeGeometry {
+    DecorativeTorus& setResolution(Real r)     { DecorativeGeometry::setResolution(r);    return *this; }
+    DecorativeTorus& setScaleFactors(const Vec3& s) { DecorativeGeometry::setScaleFactors(s); return *this; }
+    DecorativeTorus& setColor(const Vec3& rgb) { DecorativeGeometry::setColor(rgb);       return *this; }
+    DecorativeTorus& setOpacity(Real o)        { DecorativeGeometry::setOpacity(o);       return *this; }
+    DecorativeTorus& setLineThickness(Real t)  { DecorativeGeometry::setLineThickness(t); return *this; }
+    DecorativeTorus& setRepresentation(const Representation& r)
+    {
+        DecorativeGeometry::setRepresentation(r); return *this;
+    }
+
+    SimTK_PIMPL_DOWNCAST(DecorativeTorus, DecorativeGeometry);
+private:
+    class DecorativeTorusRep& updRep();
+    const DecorativeTorusRep& getRep() const;
+};
+
+
+/** An arrow with origin point, direction and length. Note that the actual 
+placement can be changed by the parent class transform & scale; here we are 
+just generating the initial arrow in the geometry object's local frame.
+
+There is a default constructor for this object but it is not much
+use unless followed by point, direction and length specifications. By default 
+we produce an arrow going from (0,0,0) in direction (1,1,1) of length 1 just 
+so it will show up if you forget to set it to something meaningful. Having a 
+default constructor allows us to have arrays of these objects. **/
+class SimTK_SimTKCOMMON_EXPORT DecorativeArrow : public DecorativeGeometry {
+public:
+    explicit DecorativeArrow(const Vec3& p1 = Vec3(0), const Vec3& dir=Vec3(1), Real length=1.0); // Arrow 
+    const Vec3& getOrigin() const;
+    const Vec3& getDirection() const;
+    const Real& getLength() const;
+
+    DecorativeArrow& setOrigin(const Vec3& origin);
+    DecorativeArrow& setDirection(const Vec3& direction);
+    DecorativeArrow& setLength(Real& length);
+
+    // Retain the derived type when setting generic geometry options.
+    DecorativeArrow& setBodyId(int b)          { DecorativeGeometry::setBodyId(b);        return *this; }
+    DecorativeArrow& setIndexOnBody(int x)     { DecorativeGeometry::setIndexOnBody(x);   return *this; }
+    DecorativeArrow& setUserRef(void* p)       { DecorativeGeometry::setUserRef(p);       return *this; }
+    DecorativeArrow& setTransform(const Transform& X_BD) { DecorativeGeometry::setTransform(X_BD); return *this; }
+    DecorativeArrow& setResolution(Real r)     { DecorativeGeometry::setResolution(r);    return *this; }
+    DecorativeArrow& setScaleFactors(const Vec3& s) { DecorativeGeometry::setScaleFactors(s); return *this; }
+    DecorativeArrow& setColor(const Vec3& rgb) { DecorativeGeometry::setColor(rgb);       return *this; }
+    DecorativeArrow& setOpacity(Real o)        { DecorativeGeometry::setOpacity(o);       return *this; }
+    DecorativeArrow& setLineThickness(Real t)  { DecorativeGeometry::setLineThickness(t); return *this; }
```

Comment:
If information is to be conveyed by the thickness of the arrow shaft (I have used logarithmic thickness to represent large values rather than making a very, very long line), will line thickness be sufficient (this maps to OpenGL's line thickness support)? Or will you want to draw a cylinder for the shaft?